sbi/neural_nets/density_estimators/flow.py

```diff
@@ -55,9 +57,52 @@ def sample(self, sample_shape: Shape, condition: Tensor) -> Tensor:
 
         num_samples = torch.Size(sample_shape).numel()
 
-        # nflows.sample() expects conditions to be batched.
         if len(condition.shape) == 1:
+            # nflows.sample() expects conditions to be batched.
+            condition = condition.unsqueeze(0)
```

Comment:
this will break if the condition is batched, right? The `Posterior` in `sbi` allows for the `context` to be batched, and I would like for the `DensityEstimator` to follow the API of the `Posterior`s.

Reply:
sorry, misunderstood, please disregard above comment

Reply:
If the condition is batched len(condition.shape) > 1, no?
It only breaks if you batch 0d tensors (which is not supported, or?)

Reply:
but what if it is an unbatched image? It would not unsqueeze and then break, no? Or do we assuem that the density estimator always gets 1D data?

Reply:
We are handling this case in our current setup already, no? 
If `x_o` is as 2D image, we automatically batch it to `(1, n, n)` using `process_x` and then pass it to the `flow.sample(num_samples, condition)`

Reply:
We cannot assume that the DensityEstimator always gets 1D, because the embedding net will be part of it, no? 
The `DensityEstimator` cannot know whether a `(N, M)`-shaped condition is an image, or a batch of vectors. We only know that upstream by knowing `x_shape`. 
We should document this well in the `DensityEstimator` and then make sure we always pass `condition` with batch dimension (what we are currently doing).